__tests__/BookmarkButton-test.js

```diff
@@ -4,54 +4,110 @@ import BookmarkButton from '@screens/app/Components/BookmarkButton/BookmarkButto
 });
 
-describe('testing icon button bookmark values', () => {
+function testPropTypesWithoutErrors(isBookmarked) {
   const bookmark = 'bookmark';
   const bookmarkOutline = 'bookmark-outline';
-  const falseWrapper = shallow(<BookmarkButton isBookmarked={false} />);
-  const falseIconButton = falseWrapper.find(IconButton).at(0);
-  it('bookmark outlined when isBookmarked is false', () => {
-    const bookmarkValue = falseIconButton.props().icon;
-    expect(bookmarkValue).toEqual(bookmarkOutline);
-  });
-  const trueWrapper = shallow(<BookmarkButton isBookmarked={true} />);
-  const trueIconButton = trueWrapper.find(IconButton).at(0);
-  it('bookmark filled when isLiked is true', () => {
-    const bookmarkValue = trueIconButton.props().icon;
-    expect(bookmarkValue).toEqual(bookmark);
-  });
-  const undefinedWrapper = shallow(<BookmarkButton isBookmarked={undefined} />);
-  const undefinedIconButton = undefinedWrapper.find(IconButton).at(0);
-  it('bookmark filled when isLiked is undefined ', () => {
-    const bookmarkValue = undefinedIconButton.props().icon;
-    expect(bookmarkValue).toEqual(bookmarkOutline);
+  const colorBlack = 'black';
+  const colorGrey = 'grey';
+  it(
+    'testing icon button heart values (No errors) where isBookmarked is ' +
+      isBookmarked,
+    () => {
+      const wrapper = shallow(<BookmarkButton isBookmarked={isBookmarked} />);
+      const iconButton = wrapper.find(IconButton).at(0);
+      const iconBookmarkedValue = iconButton.props().icon;
+      const iconColorValue = iconButton.props().color;
+      if (isBookmarked === true) {
+        expect(iconBookmarkedValue).toEqual(bookmark);
+        expect(iconColorValue).toEqual(colorBlack);
+      } else if (isBookmarked === false) {
+        expect(iconBookmarkedValue).toEqual(bookmarkOutline);
+        expect(iconColorValue).toEqual(colorGrey);
+      }
+    },
+  );
+}
+
+function testPropTypesWithErrors(isBookmarked, errorMsg, testDescription) {
+  it('Throws failed propType Error for ' + testDescription, () => {
+    let result = checkPropTypes(
+      BookmarkButton.propTypes,
+      {isBookmarked: isBookmarked},
+      'prop',
+      BookmarkButton.isBookmarked,
+    );
+    console.log(result);
+    assert(result === errorMsg);
   });
+}
+
+describe('testing icon button bookmarked values', () => {
+  testPropTypesWithoutErrors(true);
+  testPropTypesWithoutErrors(false);
+  testPropTypesWithErrors(
+    undefined,
+    'Failed prop type: The prop `isBookmarked` is marked as required in `<<anonymous>>`, but its value is `undefined`.',
+    'undefined value given to isBookmarked',
+  );
+  testPropTypesWithErrors(
+    null,
+    'Failed prop type: The prop `isBookmarked` is marked as required in `<<anonymous>>`, but its value is `null`.',
+    'null value given to isBookmarked',
+  );
+  testPropTypesWithErrors(
+    '',
+    'Failed prop type: Invalid prop `isBookmarked` of type `string` supplied to `<<anonymous>>`, expected `boolean`.',
```

Comment:
Just a side note, odd that me and you get different error messages (Possibly we should look into why that is in the future)

Reply:
Yes, we should!